controllers/defaults/placements.go

```diff
@@ -7,9 +7,12 @@ import (
 )
 
 var (
-	// osdPvcLabelSelector is the common key in prepare and OSD pod. Used
+	// osdLabelSelector is the key in OSD pod. Used
 	// as a label selector for topology spread constraints.
-	osdPvcLabelSelector = "ceph.rook.io/pvc"
+	osdLabelSelector = "rook-ceph-osd"
+	// osdPrepareLabelSelector is the key in OSD prepare pod. Used
+	// as a label selector for topology spread constraints.
+	osdPrepareLabelSelector = "rook-ceph-osd-prepare"
```

Comment:
There is a corner case to consider... If the prepare jobs have been deleted, and more OSDs are added, if the two TSCs are completely independent, the prepare jobs won't really know where to schedule evenly. 

Notice in the [upstream example](https://github.com/rook/rook/blob/master/deploy/examples/cluster-on-pvc.yaml#L94-L102), there are two conditions in the prepare job TSC that will look for both OSD daemons and prepare pods. The same thing could be accomplished if the prepare job TSC still used the selector `ceph.rook.io/pvc`, instead of having two conditions. 
```suggestion
	osdPrepareLabelSelector = "ceph.rook.io/pvc"
```

Reply:
I see on ocs operator we have some complicated logic to check if the server version is greater than some version we apply the TSC or else we apply the pod anti affinities, considering the above hacks were added a long way back when TSC were a new thing and now TSC are supported in the kube versions shouldn't we just remove the complicated logic that we have today and use simple logic like rook upstream uses?
https://github.com/red-hat-storage/ocs-operator/blob/742f979079f3eb67f74311a4838cd4d64619f554/controllers/storagecluster/cephcluster.go#L752

Reply:
Agreed, we can just remove those checks and assume we will always use TSCs. That was checking for K8s 1.19, which we are certainly always have much higher than that version now.

Reply:
Updated now to exactly mirror rook upstream, 
https://github.com/rook/rook/blob/f1aa4f37cfcc561da263f546fb3ba32dde183572/deploy/examples/cluster-on-pvc.yaml#L77-L114